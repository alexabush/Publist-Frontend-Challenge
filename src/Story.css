.Story {
  max-width: 400px;
  margin: 10px;
  border: 1px solid black;
}

.Story__content {
  padding: 0 20px;
}

.Image__container {
  width: 400px;
  height: 200px;
  overflow: hidden;
}

.Image__container img {
  width: 100%;
}

.Story__div--row {
  display: flex;
  justify-content: space-between;
}
