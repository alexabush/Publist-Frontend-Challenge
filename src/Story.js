import React, { Component } from 'react';
import './Story.css';

class Story extends Component {
  state = {};
  render() {
    const {
      image_url,
      url,
      title,
      description,
      read_time
    } = this.props.storyData;
    const score = Math.round(this.props.storyData.score);
    return (
      <div className="Story">
        <div className="Image__container">
          <img src={image_url} alt="" />
        </div>
        <div className="Story__content">
          <h2>{title}</h2>
          {/* <p className="description">{description}</p> */}
          <div className="Story__div--row">
            <p>Score: {score}</p>
            <p>{this.props.storyData.site.name}</p>
            <p>Read Time: {read_time}s</p>
          </div>
        </div>
      </div>
    );
  }
}

export default Story;
