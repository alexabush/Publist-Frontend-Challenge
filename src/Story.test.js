import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';
import Story from './Story';

const fakeData = {
  url: '',
  title: 'Google, Facebook, Apple, Amazon may pay more to operate in Sunnyvale',
  site: {
    name: 'The Mercury News'
  },
  score: '80',
  read_time: '120',
  image_url:
    'https://www.mercurynews.com/wp-content/uploads/2018/06/SJM-L-SUNNYTAX-0607-0.jpg'
};

// describe('Login Component', () => {
//   // make our assertion and what we expect to happen
//   it('should render without throwing an error', () => {
//     expect(shallow(<Story storyData={fakeData} />)).toBe(true);
//   });
// });

test('Link changes the class when hovered', () => {
  const component = renderer.create(<Story storyData={fakeData} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// it('renders correctly', () => {
//   const tree = renderer.create(<Story storyData={fakeData} />).toJSON();
//   expect(tree).toMatchSnapshot();
// });
