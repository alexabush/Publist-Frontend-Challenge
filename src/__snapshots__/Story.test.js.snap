// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="Story"
>
  <div
    className="Image__container"
  >
    <img
      alt=""
      src="https://www.mercurynews.com/wp-content/uploads/2018/06/SJM-L-SUNNYTAX-0607-0.jpg"
    />
  </div>
  <div
    className="Story__content"
  >
    <h2>
      Google, Facebook, Apple, Amazon may pay more to operate in Sunnyvale
    </h2>
    <div
      className="Story__div--row"
    >
      <p>
        Score: 
        80
      </p>
      <p>
        The Mercury News
      </p>
      <p>
        Read Time: 
        120
        s
      </p>
    </div>
  </div>
</div>
`;
